sos/plugins/openstack_instack.py

```diff
@@ -28,6 +30,16 @@ def setup(self):
         self.add_copy_spec("/home/stack/instackenv.json")
         self.add_copy_spec("/home/stack/undercloud.conf")
 
+        # collect undercloud-debug: undercloud-debug.log will be created so we
+        # need to generate it in a tmp.dir, collect it now and then cleanup it
+        tmpdir = gettempdir()
+        self.get_cmd_output_now("su -s /bin/sh -c 'source ~/stackrc && " +
+                                "/usr/libexec/openstack-tripleo/" +
+                                "undercloud-debug' stack",
+                                timeout=600, runat=tmpdir,
+                                suggest_filename="undercloud-debug.log")
+        remove("%s/undercloud-debug.log" % tmpdir)
```

Comment:
IMHO it would be better to just stick to the tripleo specific commands from undercloud-debug [1] such as the `heat stack-list` loop. The various other commands can be and mostly are already collected in their specific openstack plugin as they are not unique to an undercloud.

[1] https://github.com/openstack/tripleo-incubator/blob/master/scripts/undercloud-debug